### PR TITLE
[AP-3963] Add file upload functionality

### PR DESCRIPTION
**Pull request by jsugarman**
## What
Add file upload functionality

[Link to story](https://dsdmoj.atlassian.net/browse/AP-3963)

In line with requirements the case workers are to upload a
file containing the details of the Legal Aid Applicants details
for whom they want to retrieve details from HMRC.

## TODO
 - [X] add active storage migrations
 - [X] add "bulk_submissions" | "hmrc_bulk_submissions" table/migration
 - [X] add bulk_submissions | hmrc_bulk_submissions model
 - [X] add s3 config
 - [X] add has_one_attached :original_file to "bulk_submission"
 - [X] validations on attached file (content_type, file size, acceptable headers and column order)
 - [x] add non-js file upload forms, controller actions routes
 - [x] working on deployed environment
 - [x] request specs
 - [x] system specs
 - [x] unit specs
 - [x] amend default activestorage controllers to be authenticated or remove
         https://edgeguides.rubyonrails.org/active_storage_overview.html#authenticated-controllers or remove them entirely 
 - [x] ~~add JS file upload enhancement (dropzone), controller actions routes~~ [Separate ticket](https://dsdmoj.atlassian.net/browse/AP-4035)

## Checklist

Before you ask people to review this PR:

- Tests and linters should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/forms/bulk_submission_form.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+           :file_too_big,
+           :file_content_type
+
+  def self.max_file_size
+    MAX_FILE_SIZE
+  end
+
+  # Files already uploaded and created with bulk submissions associations
+  # so we do not need to save anything at this point, only to validate that
+  # there is one and only one file
```
Comment (jsugarman) [inline app/forms/bulk_submission_form.rb:27]: remove

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 class ApplicationController < ActionController::Base
   default_form_builder(GOVUKDesignSystemFormBuilder::FormBuilder)
-
+  # protect_from_forgery prepend: true
```
Comment (jsugarman) [inline app/controllers/application_controller.rb:3]: remove or apply and test (manually)

Comment (jsugarman) [reply app/controllers/application_controller.rb:3]: removed as [config.action_controller.default_protect_from_forgery](https://edgeguides.rubyonrails.org/configuring.html#config-action-controller-default-protect-from-forgery) is true from version 5.2 onwards so is unnecessary

**Review by kmahern — APPROVED** [review]
LGTM

**Pull request merged**